docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx

```diff
@@ -0,0 +1,428 @@
+
+Adjust the commands to match your configuration.
+
+</Admonition>
+
+Teleport uses mutual TLS authentication with self-hosted databases. These databases must be configured with Teleport's certificate authority to be able to verify client certificates.
+
+Login to Teleport on your machine.
+
+```code
+$ tsh login --proxy=<Var name="example.teleport.sh" /> --user=alice
+```
+
+Export the database client CA and copy it to target machine.
+
+```code
+# Export Teleport's database client certificate authority
+$ tctl auth export --type=db-client > teleport-db-client-ca.crt
+# Copy the CA to the Oracle Exadata VM
+$ scp teleport-db-client-ca.crt opc@demodb-vm:/tmp/teleport-db-client-ca.crt
+```
+
+Login to the target machine and switch to `grid` user.
+
+```code
+# Connect to the VM
+$ ssh opc@demodb-vm
+# Switch to grid user
+$ sudo su - grid
+```
+
+Update the TCPS wallet to trust the Teleport User Database CA.
+
+```code
+# Read the wallet password
+$ mkstore -wrl /u01/app/oracle/admin/cprops/cprops_wallet -nologo -viewEntry grid_tcps_wallet_passwd
+grid_tcps_wallet_passwd = <wallet password>
+# Update the grid TCPS wallet; provide the password when prompted
+$ orapki wallet add -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -trusted_cert -cert /tmp/teleport-db-client-ca.crt
+```
+
+Enable TLS auth for grid listener by editing the configuration.
+
+```code
+# Edit listener.ora to read
+# SSL_CLIENT_AUTHENTICATION = TRUE
+$ vi $ORACLE_HOME/network/admin/listener.ora
+...
+```
+
+For each database, update the configuration in the database home. Assuming a single database called `spark`:
+
+```code
+# Switch to oracle user
+$ sudo su - oracle
+# Load env file for database 'spark'
+$ . spark.env 
+# Edit listener.ora to read
+# SSL_CLIENT_AUTHENTICATION = TRUE
+# SQLNET.AUTHENTICATION_SERVICES = (ALL)
+$ vi $ORACLE_HOME/network/admin/sqlnet.ora
+...
+```
+
+Repeat for additional databases as needed.
+
+Once finished, restart the listener using the `grid` user.
+
+```code
+$ lsnrctl stop
+$ lsnrctl start
+```
+
+Additionally, your Oracle Database user accounts must be configured to require a valid client certificate. If you're creating a new user:
+
+```sql
+CREATE USER alice IDENTIFIED EXTERNALLY AS 'CN=alice';
+GRANT CREATE SESSION TO alice;
+```
+
+## Step 3/6. Configure Teleport.
+
+Export the database certificate from Oracle. Teleport will expect this certificate when connecting to the database.
+
+```code
+# Connect to the VM
+$ ssh opc@demodb-vm
+# Switch to grid user
+$ sudo su - grid
+# Export the database certificate. Update the "-dn" parameter to match your actual setup.
+$ orapki wallet export -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -dn "CN=demodb-vm-scan.exadatadomain.oke.oraclevcn.com"  -cert /tmp/oracle-server-certificate.crt
+```
+
+Save the `/tmp/oracle-server-certificate.crt` file to a suitable location. It will be used by the Database service when connectiong to the Oracle database. 
```

Comment:
Same question as above essentially, I may be missing some implementation details of how Oracle auth works but this makes it sound like we're going to use this certificate for connecting to the database rather than as a CA to verify cert presented by the database.

Reply:
Agreed, now that you've pointed it out I can see how it can be read in a different way than the one I've intended. I'll rephrase.